tests/components/group/test_init.py

```diff
@@ -18,307 +17,375 @@
+    """Test if the group turns off.
+
+    If no states existed and now a state it is tracking is being added
+    as OFF.
+    """
+    assert await async_setup_component(hass, "group", {})
+    test_group = await group.Group.async_create_group(
+        hass, "test group", ["light.not_there_1"]
+    )
+
+    hass.states.async_set("light.not_there_1", STATE_OFF)
+
+    await hass.async_block_till_done()
+
+    group_state = hass.states.get(test_group.entity_id)
+    assert STATE_OFF == group_state.state
+
+
+async def test_groups_get_unique_names(hass):
+    """Two groups with same name should both have a unique entity id."""
+
+    assert await async_setup_component(hass, "group", {})
+
+    grp1 = await group.Group.async_create_group(hass, "Je suis Charlie")
+    grp2 = await group.Group.async_create_group(hass, "Je suis Charlie")
+
+    assert grp1.entity_id != grp2.entity_id
+
+
+async def test_expand_entity_ids_expands_nested_groups(hass):
+    """Test if entity ids epands to nested groups."""
+
+    assert await async_setup_component(hass, "group", {})
+
+    await group.Group.async_create_group(
+        hass, "light", ["light.test_1", "light.test_2"]
+    )
+    await group.Group.async_create_group(
+        hass, "switch", ["switch.test_1", "switch.test_2"]
+    )
+    await group.Group.async_create_group(
+        hass, "group_of_groups", ["group.light", "group.switch"]
+    )
+
+    assert [
+        "light.test_1",
+        "light.test_2",
+        "switch.test_1",
+        "switch.test_2",
+    ] == sorted(group.expand_entity_ids(hass, ["group.group_of_groups"]))
+
+
+async def test_set_assumed_state_based_on_tracked(hass):
+    """Test assumed state."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
+
+    assert await async_setup_component(hass, "group", {})
+
+    test_group = await group.Group.async_create_group(
+        hass, "init_group", ["light.Bowl", "light.Ceiling", "sensor.no_exist"]
+    )
+
+    state = hass.states.get(test_group.entity_id)
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    hass.states.async_set("light.Bowl", STATE_ON, {ATTR_ASSUMED_STATE: True})
+    await hass.async_block_till_done()
+
+    state = hass.states.get(test_group.entity_id)
+    assert state.attributes.get(ATTR_ASSUMED_STATE)
+
+    hass.states.async_set("light.Bowl", STATE_ON)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(test_group.entity_id)
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+
+async def test_group_updated_after_device_tracker_zone_change(hass):
+    """Test group state when device tracker in group changes zone."""
+    hass.states.async_set("device_tracker.Adam", STATE_HOME)
+    hass.states.async_set("device_tracker.Eve", STATE_NOT_HOME)
+    await hass.async_block_till_done()
+
+    assert await async_setup_component(hass, "group", {})
+    assert await async_setup_component(hass, "device_tracker", {})
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)